### DISCUSSION TITLE
Windows Terminal Preview v1.15.186

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#13443] [Announcements]
It's Summer in the US, which means that ~~it's really hot~~ there's a new Terminal preview release!

Here's what's in it:

<details>
<summary>Why are there so many packages? How do I choose?</summary>
This version of Windows Terminal is distributed in two bundles, one of which works on Windows 10-11 and the other of
which only works on Windows 11. The Windows 11 version is much smaller because we no longer need to work around a
platform issue related to our dependencies.

If you intend on using Terminal as an unpackaged application--that is, extracting the `msix` file--we recommend that
you use the `Win10` bundle. You will need the [Visual C++ runtime
redistributable](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist).

In addition, if you install the _packaged_ version on either Windows 10 or Windows 11, it now depends on [the Visual C++
Universal Runtime
Package](https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/cpp/libraries/c-runtime-packages-desktop-bridge?msclkid=3e1dd82aac8b11ec8f9dbd86d5979ca8#how-to-install-and-update-desktop-framework-packages).

Despite these distributions having different version numbers, they are built from the same code and there is no
functional difference between them.

If you install the Windows 10 verison on Windows 11, it will probably automatically upgrade itself to the Windows 11
version. It turns out that it is impossible to have two bundles with _the same version number_, so it has to be this
way.
</details>

## Features
* Selecting text in the terminal just got better!
   * Use the `markMode` action to enter mark mode and create a selection at the cursor (#13053) (#13358)
      * This is bound to <kbd>Ctrl+Shift+M</kbd> by default. Be sure to try it out!
   * Selections made with the keyboard now display a selection marker UI (#10865)
   * Use the `switchSelectionEndpoint` action to switch which endpoint you are moving in a selection (#13370)
   * Use the `toggleBlockSelection` action to transform your existing selection into a block selection (#13219)
* [Experimental] We now support scrollbar marks! (#12948) (#13163) (#13291) (#13414) 
   * Use the `addMark` action to add a scrollbar mark
      * The `color` optional parameter can be used to specify a color
   * Use the `scrollToMark` action with a specified `direction` parameter to scroll between the marks
   * Use the `clearMark` action to remove a selected mark
   * Use the `clearAllMarks` action to remove all scrollbar marks
   * The `experimental.autoMarkPrompts` global setting can be set to `true` to automatically mark each prompt
      * NOTE: This uses the FTCS_PROMPT sequence from FinalTerm, `OSC 133 ; A`, which we now support! (#13163)
   * The `experimental.showMarksOnScrollbar` global setting can also be set to `true` to display the marks on your scrollbar
* If you're new to Windows Terminal Preview, but already have Windows Terminal installed and customized, we now migrate your settings over (#12907) (thanks @huiyooumich!)
* The tab's context menu now has "Find" as an option (#13055) (thanks @Predelnik!)

## Changes
* "Open settings file" commands now explicitly mention "JSON" for easier searching (#13265)
* Color schemes now support `"purple"` and `"magenta"` interchangeably in the JSON (#13261) (thanks @matthewd673!)
* An accelerator key is now defined for the "Open in Terminal" shell extension (#13080) (thanks @ianjoneill!)
* The settings UI's "Save" pane now aligns with the "Open JSON file" footer (#13282) (thanks @HO-COOH!)
* The Default Terminal setting in settings UI now has a "Let Windows decide" option (#13160)
   * An occasional crash while opening the settings UI has been stomped out (same PR!)
* The "Save" and "Discard changes" buttons were reordered in the settings UI to more closely follow the Windows UI guidelines (#13237)
* @dansmor7 has refined how colored tabs look when they're out of focus or hovered (#13434) (thanks!)

### More Escape Sequences and expanded VT support

_Courtesy of @j4james_:

* Applications can now use `DECCTR` to alter the terminal's color scheme (#13139) (#13227)
* The same applications can now use `DECAC` to assign a color to the default foreground and background colors, as well as change the tab background color (#13058)
* Other applications can now use `DECPS` to play a basic sequence of musical notes (#13208)
   * This feature is preview-only until we can make sure the MIDI sound font is available everywhere Stable ships.

### Documentation
* building.md and mouseInput.cpp got cleaned up a bit (#13333) (thanks @ofek and @oferze!)
* We added a Gannt chart to the roadmap (#13234)

## Bug Fixes
* We no longer suppress black blackground or gray foreground for PowerShell (#13352)
   * We have chosen to remove this workaround as newer versions of PowerShell's PSReadline component contain a fix for the issue.
   * This was a compatibility band-aid that was impacting the capabilities of great projects such as [Oh My Posh](https://ohmyposh.dev).
   * ❗ If you see unexpected black backgrounds appearing behind text while typing a command in PowerShell, make sure your PSReadline version is up to date. You can update your version of PSReadline by running the command, `Update-Module PSReadline`.
* The Default Terminal banner is now hidden if you opened a session via default terminal (#13344)
   * AKA: We won't nag you to set Terminal as your default if it's demonstrably the default ;P
* `[O` is no longer output erroneously from focus events for clients of libuv like neovim (#13260)
* AtlasEngine no longer _secretly_ increases the font size of HTML/RTF copies when the font changes (#13384)
* Keyboard selection is now limited to the scrollable area (#13318)
* The "Open in Terminal" shell extension is now hidden when accessing a non-filesystem path like "Quick Actions" (#13206) (thanks @leejy12!)
* Clearing the screen via `cls` or `Clear-Host` won't leave behind an erroneous line of text (#13324) (thanks @j4james!)
* Default Terminal sessions now properly pass focus events when opened (#13247)
* Terminal will now use Unicode 14.0 to determine the width of some Unicode characters (#13292)
* We will no longer try to launch `wsl` to ask it to tell us about distributions when it's obvious that you don't have any (#13436)
* We've fixed a minor race condition in default terminal handoff that impacted nobody (#13410)

### Reliability
* We no longer crash when a screen reader is reading from a CLI app using the alt buffer (#13250) (#13244)
* Deleting the last profile in the settings UI no longer causes a crash (#13242)
* Opening Windows Terminal via the Win+X menu no longer occasionally crashes (#13212)
* `SetConsoleWindowInfo` can no longer crash a terminal tab (#13212)

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.15.1862.0'>Windows Terminal Preview v1.15.186</a>.</em>

**Comment by afkvido** [comment]
cool

**Comment by yunqinglt** [comment]
帅！

**Comment by TBBle** [comment]
The auto-mark prompts thing sounds really cool (e.g., for "scroll to start of whatever verbose output I'm staring at" purposes), and I just had a skim of #11000 and the related PRs.

Specifically on
> * The experimental.autoMarkPrompts global setting can be set to true to automatically mark each prompt
>   * NOTE: This uses the FTCS_PROMPT sequence from FinalTerm, OSC 133 ; A, which we now support! #13163

I was curious about this, and it seems from the code that this pair of dot-points has conflated two things? I haven't _tried_ it yet (I'm on my non-Terminal-preview machine) but it looks like `FTCS_PROMPT` works always, while `experimental.autoMarkPrompts` (which is a per-session setting ~~AFAICT~~ see #13583) is only used for the heuristic prompt-marking on `vkey == VK_RETURN && !_inAltBuffer()`.

**Reply by zadjii-msft** [reply]
Meh, yes, those are two related but different things. Ultimately, the `experimental.autoMarkPrompts` create a "prompt" mark, in the same way that `FTCS_PROMPT` would. On the backend, we treat them both as the same thing in 1.15.

**Comment by XPythonista** [comment]
This terminal looks good in trailer not in reality

**Reply by zadjii-msft** [reply]
Do you have some actionable feedback to provide?

**Reply by MadTinker** [reply]
He's twelve. My terminal is beautiful, thank you everyone.

**Comment by mgjuranic** [comment]
I get a following error while trying to install it:

Add-AppPackage : Deployment failed with HRESULT: 0x80073CF3, Package failed updates, dependency or conflict validation.
Windows cannot install package Microsoft.WindowsTerminalPreview_1.15.1862.0_x64__8wekyb3d8bbwe because this package dep
ends on a framework that could not be found. Provide the framework "Microsoft.Midi.GmDls" published by "CN=Microsoft Co
rporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", with neutral or x64 processor architecture and mini
mum version 1.0.0.0, along with this package to install. The frameworks with name "Microsoft.Midi.GmDls" currently inst
alled are: {}
NOTE: For additional information, look for [ActivityId] 6518c759-9824-0001-dc05-19652498d801 in the Event Log or use th
e command line Get-AppPackageLog -ActivityID 6518c759-9824-0001-dc05-19652498d801
At line:1 char:1
+ Add-AppPackage .\Microsoft.WindowsTerminalPreview_Win10_1.15.1862.0_8 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\Users\mgjura...bbwe.msixbundle:String) [Add-AppxPackage], IOException
    + FullyQualifiedErrorId : DeploymentError,Microsoft.Windows.Appx.PackageManager.Commands.AddAppxPackageCommand

**Reply by TBBle** [reply]
That's possibly the reason for
> This feature is preview-only until we can make sure the MIDI sound font is available everywhere Stable ships.

I'm not sure how you install that `Microsoft.Midi.GmDls` package manually, though. Presumably if you were installing from the Windows Store, it'd have installed that package automatically. It _looks_ like the Windows 10 SDKs include it in a file named "MidiVoices.appx", if that helps.

**Reply by zadjii-msft** [reply]
You may also want to refer to #13520
